File: problemas/1000/1049.dart
import 'dart:io';

void main() {
  String palavra = stdin.readLineSync();
  String animal;

  if (palavra == 'vertebrado')
  {
    palavra = stdin.readLineSync();
    if (palavra == 'ave')
    {
      palavra = stdin.readLineSync();
      animal = palavra == 'carnivoro' ? 'aguia' : 'pomba';
    }
    else
    {
      palavra = stdin.readLineSync();
      animal = palavra == 'onivoro' ? 'homem' : 'vaca';
    }
  }
  else
  {
    palavra = stdin.readLineSync();
    if (palavra == 'inseto')
    {
      palavra = stdin.readLineSync();
      animal = palavra == 'hematofago' ? 'pulga' : 'lagarta';
    }
    else
    {
      palavra = stdin.readLineSync();
      animal = palavra == 'hematofago' ? 'sanguessuga' : 'minhoca';
    }
  }

  stdout.writeln(animal);
}

File: problemas/1000/1049.js
const entrada = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
palavra = entrada.shift();

if (palavra === 'vertebrado') {
  palavra = entrada.shift();
  if (palavra === 'ave') {
    palavra = entrada.shift();
    animal = palavra === 'carnivoro'  ? 'aguia' : 'pomba';
  } else {
    palavra = entrada.shift();
    animal = palavra === 'onivoro'  ? 'homem' : 'vaca';
  }
} else {
  palavra = entrada.shift();
  if (palavra === 'inseto') {
    palavra = entrada.shift();
    animal = palavra === 'hematofago' ? 'pulga' : 'lagarta';
  } else {
    palavra = entrada.shift();
    animal = palavra === 'hematofago' ? 'sanguessuga' : 'minhoca';
  }
}

console.log(animal);